sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/_patch.py

```diff
@@ -75,6 +78,17 @@ def __init__(
         ledger_certificate_path: Union[bytes, str, os.PathLike],
         **kwargs: Any,
     ) -> None:
+        # Remove some kwargs first so that there aren't unexpected kwargs passed to
+        # get_ledger_identity.
+        if isinstance(credential, ConfidentialLedgerCertificateCredential):
+            auth_policy = None
+        else:
+            credential_scopes = kwargs.pop("credential_scopes", ["https://confidential-ledger.azure.com/.default"])
+            auth_policy = kwargs.pop(
```

Comment:
Do you mean "credential" is optional?

Reply:
Some kind of credential is always required, but an authentication policy is optional.
1. `authentication_policy` has to be popped before the later call to `identity_service_client.get_ledger_identity` because it is an unexpected kwarg for that call.
2. `auth_policy` is only `None` `if isinstance(credential, ConfidentialLedgerCertificateCredential)`. However, later we add `connection_cert` as a kwarg `if isinstance(credential, ConfidentialLedgerCertificateCredential)`. When such a credential is used, `connection_cert` is used for authentication instead of the traditional authentication policy.

Reply:
It looks to me if user specifies `authentication_policy`, they don't need to provide `credential`

Reply:
From what I can see - if I were to try specifying both a ConfidentialLedgerCertificateCredential object, and an auth policy - none of the latter kwargs would be popped and it would be ignored.
However, it seems that this was always the case prior to this PR, so I doubt it's a problem. We don't really document specifying whole policies at the constructor level, so I doubt many people are using features this advanced.